### PR TITLE
Define main classes

### PR DESCRIPTION
**Pull request by GovoniT**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -127,3 +127,5 @@ dmypy.json
 
 # Pyre type checker
 .pyre/
+# Ignore the test temp.py file
```
Comment (aidanjungo) [inline .gitignore:130]: These comment could be removed now.

File: src/gmshairfoil2d/geometry_def.py [hunk 2]
```diff
@@ -0,0 +1,88 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""
+Created on Thu Mar  3 12:02:13 2022
+
+@author: tony
+"""
+# ------------------------------------------------------------------------------
```
Comment (aidanjungo) [inline src/gmshairfoil2d/geometry_def.py:8]: This comment line is not useful, the docstring is enough

File: src/gmshairfoil2d/geometry_def.py [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+            self.curveloopTags = [curveloopTags]  # put in a array
+        else:
+            self.curveloopTags = curveloopTags  # it is already an array
+        # PlaneSurface object are of dim 2
+        self.dim = 2
+        # create the gmsh object and store the tag of the geometric object
+        self.tag = gmsh.model.occ.addPlaneSurface(self.curveloopTags)
+
+
+# end
```
Comment (aidanjungo) [inline src/gmshairfoil2d/geometry_def.py:87]: Could be removed

File: src/gmshairfoil2d/geometry_def.py [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+
+class PlaneSurface:
+    """
+    A plane surface defined by one or more curve loops tags
+    The first curve loop defines the exterior contour; additional curve loop
+    define holes in the surface domaine
+    """
+
+    def __init__(self, curveloopTags):
+        if type(curveloopTags) == int:  # if only one curve loop is used
```
Comment (aidanjungo) [inline src/gmshairfoil2d/geometry_def.py:77]: you can use:

isinstance(curvelopTag, int)

**Review by aidanjungo — COMMENTED** [review]
Check also varible case in PEP8

File: src/gmshairfoil2d/geometry_def.py [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+
+
+class PlaneSurface:
+    """
+    A plane surface defined by one or more curve loops tags
+    The first curve loop defines the exterior contour; additional curve loop
+    define holes in the surface domaine
+    """
+
+    def __init__(self, curveLoop_array):
```
Comment (aidanjungo) [inline src/gmshairfoil2d/geometry_def.py:69]: curveLoop_list instead of curveLoop_array

File: src/gmshairfoil2d/geometry_def.py [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+        self.dim = 1
+        # create the gmsh object and store the tag of the geometric object
+        self.tag = gmsh.model.occ.addLine(self.start_point.tag, self.end_point.tag)
+
+
+class CurveLoop:
+    def __init__(self, line_array):
+        # Line array must be in the order of the wanted CurveLoop and closed
+        self.line_array = line_array
+        self.tag_array = []
```
Comment (aidanjungo) [inline src/gmshairfoil2d/geometry_def.py:42]: You can use a list comprehension for that

File: src/gmshairfoil2d/geometry_def.py [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+        # tag of the two points forming the line
+        self.start_point = start_point
+        self.end_point = end_point
+        self.dim = 1
+        # create the gmsh object and store the tag of the geometric object
+        self.tag = gmsh.model.occ.addLine(self.start_point.tag, self.end_point.tag)
+
+
+class CurveLoop:
+    def __init__(self, line_array):
```
Comment (aidanjungo) [inline src/gmshairfoil2d/geometry_def.py:39]: line_list instead of line_array

File: src/gmshairfoil2d/geometry_def.py (hunk 4 above)
Comment (GovoniT) [reply src/gmshairfoil2d/geometry_def.py]: didn't know this feature, it seems indeed more appropriate when dealing with classes

**Review by aidanjungo — APPROVED** [review]

**Pull request merged**